app/Http/Controllers/PrintController.php

```diff
@@ -37,4 +69,46 @@ public function modify_balance(Request $request) {
         }
         return redirect()->route('print');
     }
-}
+
+    private function print_file($file, $cost, $two_sided, $number_of_copies) {
+        $printer_name = config('app.printer_name');
+        $state = "QUEUED";
+        try {
+            $path = $file->storeAs('', md5(rand(0, 100000) . date('c')) . '.pdf', 'printing');
+            $path = Storage::disk('printing')->getDriver()->getAdapter()->applyPathPrefix($path);
+            $result = exec("lp -d " . $printer_name
+                        . ($two_sided ? " -o sides=two-sided-long-edge " : " ") 
+                        . "-n " . $number_of_copies
+                        . $path . " 2>&1");
+            if (!preg_match("/^request id is ([^\s]*) \\(1 file\\(s\\)\\)$/", $result, $job_id)) {
+                Log::error("Printing error at line: ".__FILE__.":".__LINE__." (in function ".__FUNCTION__."). result:"
+                    . print_r($result, true));
+                $state = "ERROR";
+            }
+            $job_id = $job_id[1];
+        }
+        catch (\Exception $e){
+            Log::error("Printing error at line: ".__FILE__.":".__LINE__." (in function ".__FUNCTION__."). ". $e->getMessage());
+            $state = "ERROR";
+            $job_id = "";
```

Comment:
Why do you zero out the strings, fall through the catch and call `PrintJob::create` with invalid (possibly empty `job_id` and `cost`)?
Shouldn't you simply return on error in the catch clause?

Reply:
Ask @vendi95

Reply:
The basic idea here - for me at least - is that we need to log what happened and also save the failed printJob to find it in the history. 
From that, the user will know nothing more than some error occurred, and they don't need to know more.